widget_weather.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"fmt"
+	"image"
+	"image/color"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/muesli/streamdeck"
+)
+
+// WeatherWidget is a widget displaying the current weather.
+type WeatherWidget struct {
+	BaseWidget
+
+	location string
+	unit     string
+	color    color.Color
+}
+
+// NewWeatherWidget returns a new WeatherWidget.
+func NewWeatherWidget(bw BaseWidget, opts WidgetConfig) *WeatherWidget {
+	bw.setInterval(opts.Interval, 300000)
+
+	var location, unit string
+	_ = ConfigValue(opts.Config["location"], &location)
+	_ = ConfigValue(opts.Config["unit"], &unit)
+	var color color.Color
+	_ = ConfigValue(opts.Config["color"], &color)
+
+	return &WeatherWidget{
+		BaseWidget: bw,
+		location:   location,
+		unit:       unit,
+		color:      color,
+	}
+}
+
+// Update renders the widget.
+func (w *WeatherWidget) Update(dev *streamdeck.Device) error {
+	url := "http://wttr.in/" + w.location + "?format=%x+%t" + formatUnit(w.unit)
+	resp, err := http.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	if strings.Contains(string(body), "Unknown location") {
+		return fmt.Errorf("unknown location: %s", w.location)
+	}
+
+	wttr := strings.Split(string(body), " ")
+	cond := wttr[0]
+	temp := strings.Replace(wttr[1], "+", "", 1)
```

Comment:
We should check that `wttr` actually contains two items before accessing `wttr[1]`.

Reply:
Just did that.